### ISSUE TITLE
有转成 h5 的模型文件嘛？

### ISSUE THREAD

**Issue by airlsyn** [JunnYu/WoBERT_pytorch#1]
我尝试了 `bert4keras.model` 里的 `build_transformer_model` 转出 `h5` 后，用 `huggingface` 的 `TFBertForMaskedLM` 加载，会报 
```Some layers of TFBertForMaskedLM were not initialized from the model checkpoint at tf-chinese_wobert_plus_L-12_H-768_A-12-h5 and are newly initialized: ['mlm___cls', 'bert']```

用 TFBertModel 加载会报 ```Some layers of TFBertModel were not initialized from the model checkpoint at tf-chinese_wobert_plus_L-12_H-768_A-12-h5 and are newly initialized: ['bert']```

然后用 TFBertForMaskedLM 预测 [MASK] 时会是随机的。

**Comment by JunnYu** [comment]
`我尝试了 bert4keras.model 里的 build_transformer_model 转出 h5 后`
不能直接转成H5的格式，bert4keras转出的h5跟huggingface的h5里面权重格式保存的应该是不同的。
建议将bert4keras的模型保存成 .ckpt的格式，然后转成将ckpt格式权重转成pytorch model.bin权重，再然后TFBertForMaskedLM加载pytorch的权重

**Comment by airlsyn** [comment]
> `我尝试了 bert4keras.model 里的 build_transformer_model 转出 h5 后` 不能直接转成H5的格式，bert4keras转出的h5跟huggingface的h5里面权重格式保存的应该是不同的。 建议将bert4keras的模型保存成 .ckpt的格式，然后转成将ckpt格式权重转成pytorch model.bin权重，再然后TFBertForMaskedLM加载pytorch的权重

是的，只能曲线救国。非常感谢。

**Comment by airlsyn** [comment]
备注下（如果有其他网友遇到，又想用 h5 的），可以尝试：

```
from transformers import convert_pytorch_checkpoint_to_tf2
convert_pytorch_checkpoint_to_tf2.convert_pt_checkpoint_to_tf(
    model_type='bert',
    pytorch_checkpoint_path='pt-chinese_wobert_plus_L-12_H-768_A-12/pytorch_model.bin',
    config_file='pt-chinese_wobert_plus_L-12_H-768_A-12/config.json',
    tf_dump_path='tf_model.h5',
    compare_with_pt_model=True,
    use_cached_models=True
)
```

导出来，可以用
